pipeline/stage.go

```diff
@@ -105,3 +108,51 @@ func (s *StageSlice) Sanitize(driver string) *StageSlice {
 		return nil
 	}
 }
+
+// Empty returns true if the provided stage is empty.
+func (s *Stage) Empty() bool {
+	// return true if the container is nil
+	if s == nil {
+		return true
+	}
+
+	// return true if every stage field is empty
+	if len(s.Name) == 0 &&
+		len(s.Needs) == 0 &&
+		len(s.Steps) == 0 &&
+		len(s.Environment) == 0 {
+		return true
+	}
+
+	// return false if any of the ruletype is provided
```

Comment:
I think the comment here has a typo i.e. ~`ruletype`~

```suggestion
	// return false if any of the stage fields are not empty
```